src/hotspot/share/interpreter/bytecodeTracer.cpp

```diff
@@ -192,10 +192,13 @@ void BytecodeTracer::print_method_codes(const methodHandle& method, int from, in
   BytecodeStream s(method);
   s.set_interval(from, to);
 
-  ttyLocker ttyl;  // keep the following output coherent
```

Comment:
The previous method using ttyLocker states:
```
// The ttyLocker also prevents races between two threads
// trying to use the single instance of BytecodePrinter.
```
Is that also a concern in this method?

Reply:
FTR the code should not be assuming that `st` is `tty`!

Reply:
I was just pointing out that the code was "wrong" to assume `st` had to be the tty - though as you point out it is harmless even if not useful.

Thanks for clarifying that we don't need the ttyLocker for mutual exclusion here.